packages/web-react/docs/stories/examples/FileUploaderWithImagePreview.stories.tsx

```diff
@@ -0,0 +1,105 @@
+import FileUploaderInput from '../../../src/components/FileUploader/FileUploaderInput';
+import FileUploaderList from '../../../src/components/FileUploader/FileUploaderList';
+import FileUploaderAttachment from '../../../src/components/FileUploader/FileUploaderAttachment';
+import { Button, Modal, ModalBody, ModalDialog, ModalFooter } from '../../../src/components';
+
+export default {
+  title: 'Examples/Compositions',
+};
+
+export const FileUploaderWithModalImagePreview = (args) => {
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const [base64File, setBase64File] = React.useState<string>('');
+  const [fileToPreview, setFileToPreview] = React.useState<File | null>(null);
+  const { fileQueue, addToQueue, clearQueue, onDismiss } = useFileQueue();
+
+  const imagePreview = (key: string, file: File) => {
+    if (file.type.includes('image')) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        // FileReader.result contains the file content as a base64 encoded string
+        setBase64File(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+
+      setFileToPreview(file);
+      setIsModalOpen(true);
+
+      return;
+    }
+
+    return addToQueue(key, file);
+  };
+
+  const resetCropImageState = () => {
+    setIsModalOpen(false);
+    setFileToPreview(null);
+    setBase64File('');
+  };
+
+  const handleClose = () => {
+    resetCropImageState();
+  };
+
+  const confirmPreview = () => {
+    setIsModalOpen(false);
+    addToQueue(fileToPreview?.name || '', fileToPreview as File);
+  };
+
+  const attachmentComponent = ({ id, ...props }: SpiritFileUploaderAttachmentProps) => (
+    <FileUploaderAttachment key={id} id={id} {...props} />
+  );
+
+  return (
+    <>
+      <FileUploader
+        {...args}
+        id="fileUploaderExample"
+        onDismiss={onDismiss}
+        fileQueue={fileQueue}
+        addToQueue={imagePreview}
+        clearQueue={clearQueue}
+      >
+        <FileUploaderInput
+          id="fileUploaderExampleInput"
+          name="attachments"
+          label="Label"
+          linkText="Upload your file(s)"
+          labelText="or drag and drop here"
+          helperText="Max file size is 10 MB"
+          validationText="Validation message"
+          /* eslint-disable-next-line no-console */
+          onError={(error) => console.error('My error log', error)}
+        />
+        <FileUploaderList
+          id="fileUploaderExampleList"
+          label="Attachments"
+          inputName="attachments"
+          attachmentComponent={attachmentComponent}
+          hasImagePreview
+          isEditable
+        />
+      </FileUploader>
+
+      <Modal id="ModalExample" isOpen={isModalOpen} onClose={handleClose}>
+        <ModalDialog>
+          <ModalBody>
+            <img src={base64File} style={{ width: '100%', height: 'auto' }} alt="fileToPreview" />
+          </ModalBody>
```

Comment:
We need to be a little bit more robust here:

```suggestion
          <ModalBody>
            <div className="pt-400 pt-tablet-600">
              <img src={base64File} className="image-responsive mx-auto" alt="fileToPreview" />
            </div>
          </ModalBody>
```

This makes the image responsive using our [helper class](https://spirit.supernova-docs.io/latest/development/helpers/images-vlaxk4d0). Also, the image [offset](https://spirit.supernova-docs.io/latest/development/utilities-gxjZpDOv#section-margin-8f) will be uniform on all sides and the image will be centered if it's smaller than the Modal.